### ISSUE TITLE
Dynamically add methods, events 

### ISSUE THREAD

**Issue by parhelium** [jagi/meteor-astronomy#115]
In  astronomy@0.12.1 there were API for adding dynamically `events` and `methods`.
That feature is really useful for me.

I have created platform which contains several meteor apps and for each app there is AppName-core package which contains `models`, `validators`, `consts`, etc.
Models stored in `AppName-core` packages are very minimal, usually fields and validators and common for all apps methods

In every app I extend models of `AppName-core` packages by adding some methods, which are specific for app.

I would love to see back API which help me doing that.

- Class.addMethods
- Class.addMethod
- Class.addEvents
- Class.addEvent

Please... :)

**Comment by parhelium** [comment]
Workaround for methods is:
```
SSS.Product.prototype.someMethod =  function(){}
```

**Comment by parhelium** [comment]
Workaround for adding events:
```
SSS.Product.schema.eventManager.on( 'beforeInit', callback )
```

**Comment by parhelium** [comment]
Now I clearly see that API I need could be implemented by creating new behaviour. On the way to create my first behaviour ;-)

**Comment by banglashi** [comment]
Hi @parhelium, you can extend the class to achieve what you where doing with addMethod, addEvent...

See http://astronomy.meteor.com/#extending-class

**Comment by parhelium** [comment]
@banglashi well, you are absolutely right !
it looks like I should more focus on reading documentation rather than getting into the source code so quickly ;]

**Comment by jagi** [comment]
:-) yes I've changed that to more flexible `extend()` method. Instead of learning each method name you can pass schema extension. From 1.0.0 behaviors work the same way. I think it's a lot easier to work with :-)

**Comment by parhelium** [comment]
Great job @jagi !

**Comment by parhelium** [comment]
My workarounds work fine ( on the left ), but extending class doesn't work ;] ( on the right ) : 

<img width="1510" alt="screen shot 2015-09-30 at 09 22 39" src="https://cloud.githubusercontent.com/assets/725686/10186909/e7536562-6754-11e5-9a2d-60745d4929a7.png">

**Comment by parhelium** [comment]
It looks it is bug connected with adding methods dynamically, because events works fine:
```
SSS.ProductGroup.extend({
    events:{
        beforeInit:function(){
            console.log('SSS.ProductGroup.events.beforeInit ')
        }
    }
})
```

**Comment by jagi** [comment]
Ok I will take a look at it

**Comment by jagi** [comment]
Yes it a bug. Hmm and it appears to be something that I've omitted and can be a little bit hard to fix. However it has to be done. I'm getting to work. Thanks for noticing it. Will let you know when I'm done.

**Comment by parhelium** [comment]
Thanks ! Good luck !

**Comment by jagi** [comment]
Ok I've fixed it. I will have to make some more tests and I will release it. While making this fix I had to make many changes however it won't influence any regular user. However it may have some effect on modules developers. The fix made a code even more readable and faster in some cases :)

**Comment by parhelium** [comment]
Let me know how i can test ( maybe 1.0.1-rc-1  or 1.0.0-rc-7 ) ?

**Comment by jagi** [comment]
I will just publish 1.0.1 version today

**Comment by jagi** [comment]
Fixed in 1.0.2

**Comment by parhelium** [comment]
Great, thanks ! Will test it soon :)
